Bio/motifs/matrix.py

```diff
@@ -405,26 +405,40 @@ def calculate(self, sequence):
         else:
             return scores
 
-    def search(self, sequence, threshold=0.0, both=True):
+    def search(self, sequence, threshold=0.0, both=True, chunksize=10**6):
         """Find hits with PWM score above given threshold.
 
         A generator function, returning found hits in the given sequence
         with the pwm score higher than the threshold.
         """
+        import numpy as np
```

Comment:
@mdehoon my understanding is ``Bio.motifs`` defaults to using C code called via a numpy array, but can fall back on a pure Python no numpy alternative. If so, then the lazy import is OK.

If you regard numpy as a hard requirement of ``Bio.motifs`` then we should just have the ``import numpy`` once at top level.

Reply:
Since numpy is now a requirement of Biopython, we can remove the pure Python implementation, and have the `import numpy` once at top level.

Reply:
Done in the latest commit.

Reply:
We need to clarify the numpy situation, logged as #2390.